File: FunctionTestRunnerTS/FunctionTestRunner/src/utils/test-configuration.ts
import config from 'config';

export class TestConfiguration {
  getEnvironment(): string {
    return this.getSetting('testSetup.environment');
  }

  getApiKey(): string {
    return this.getSetting('apiKey');
  }

  getApiBaseUrl(): string {
    return this.getSetting('apiBaseUrl');
  }

  getApiTimeout(): number {
    return this.getNumberSetting('apiTimeout');
  }

  getInternalClientBaseUrl(): string {
    return this.getSetting('internalApiBaseUrl');
  }

  getSetting(settingKey: string): string {
    if (!config.has(settingKey)) {
      throw new Error(`AppSetting ${settingKey} not defined`);
    }

    const value: string = config.get(settingKey);

    return value;
  }

  getNumberSetting(settingKey: string): number {
    const value = this.getSetting(settingKey);

    return Number.parseInt(value, 10);
  }
}

File: FunctionTestRunner/FunctionTestRunner/Utils/TestConfiguration.cs
using FunctionTestRunner.Exceptions;
using Microsoft.Extensions.Configuration;

namespace FunctionTestRunner.Utils;

public abstract class TestConfiguration : ITestConfiguration
{
    protected readonly IConfigurationRoot config;

    public TestConfiguration()
    {
        config = new ConfigurationBuilder()
                .AddJsonFile("appsettings.test.json")
                .Build();
    }

    public string GetEnvironment()
    {
        return GetAppSetting("environmentToTest");
    }

    public string GetApiKey()
    {
        return GetAppSetting("apiKey");
    }

    public string GetApiBaseUrl()
    {
        return GetAppSetting("apiBaseUrl");
    }

    public string GetUsername()
    {
        return GetAppSetting("username");
    }

    public string GetPassword()
    {
        return GetAppSetting("password");
    }

    // Add more internalAPi info

    public string GetInternalApiBaseUrl()
    {
        return GetAppSetting("internalApiBaseUrl");
    }

    public string GetLoginInternalApiUrl()
    {
        return GetAppSetting("loginInternalApiUrl");
    }

    protected string GetAppSetting(string appSetting)
    {

        var value = config[appSetting];

        if (value == null)
        {
            throw new AppSettingNotDefinedException($"AppSetting {appSetting} not defined");
        }

        return value;
    }
}
